sopel/modules/weather.py

```diff
@@ -17,52 +17,50 @@ def woeid_search(query):
     node for the result, so that location data can still be retrieved. Returns
     None if there is no result, or the woeid field is empty.
     """
-    query = 'q=select * from geo.places where text="%s"' % query
-    body = requests.get('https://query.yahooapis.com/v1/public/yql?' + query)
-    parsed = xmltodict.parse(body.text).get('query')
-    results = parsed.get('results')
-    if results is None or results.get('place') is None:
+    query = query.replace(" ", "+")
+    body = requests.get('https://www.metaweather.com/api/location/search/?query=' + query)
```

Comment:
You should let `requests` do its job with the query string:

```
>>> r = requests.get('https://www.metaweather.com/api/location/search/', {'query': 'Los Angeles'})
>>> r.json()
[{'latt_long': '34.053490,-118.245323', 'woeid': 2442047, 'location_type': 'City', 'title': 'Los Angeles'}]
>>> r.url
'https://www.metaweather.com/api/location/search/?query=Los+Angeles'
```

You don't need to escape it yourself, *and* it'll take care itself of any special char.